File: 3way/problem9/solution.c
int compress(char* chars, int charsSize) {
    int count = 1;
    int index = 0;
    for(int i = 1; i <= charsSize; i++){
        if(i == charsSize || chars[i] != chars[i - 1]){
            chars[index++] = chars[i - 1];
            if(count > 1){
                char countStr[12];
                sprintf(countStr, "%d", count);
                for (int j = 0; countStr[j] != '\0'; j++) {
                    chars[index++] = countStr[j];
                }
                count = 1;
            }
        }else{
            count++;
        }
    }
    return index;
}

File: 3way/problem9/solution.js
/**
 * @param {character[]} chars
 * @return {number}
 */
var compress = function(chars) {
    let count = 1;
    let index = 0;

    for(let i = 1; i <= chars.length; i++){
        if(i == chars.length || chars[i] != chars[i - 1]){
            chars[index++] = chars[i - 1];
            if(count > 1){
                const countStr = count.toString();
                for (let c of countStr) {
                    chars[index++] = c;
                }
                count = 1;
            }
        }else{
            count++;
        }
    }
    return index;
};
